### PR TITLE
Sing on the radio with Y and sundry other adjustments

### PR DESCRIPTION
**Pull request by ausLancaster**
## About the PR 

- Singing prefix '%' will now be parsed when using radio with the Y key .
- You can still use say or T with radio prefixes and '%' for same result e.g. ";%lalala. Now "%;lalala" will work, as will "; %lalala".
- ~~Fixed bug where \<i> tags were appearing with void voice (e.g. when wearing obsidian crown).~~

## Why's this needed? 

- Y is the easiest way to use the radio. (Your author was not aware of this when he implemented singing.)
- Some people were confused about how to sing on the radio.

## Changelog

```
(u)austin:
(+)Can now use Y then prefix with '%' to sing on the radio.
```
[ENHANCEMENT]

### CODE REVIEW

**Comment by Sovexe** [comment]
> Fixed bug where <i> tags were appearing with void voice (e.g. when wearing obsidian crown).
I believe this may have already been fixed

see #2517 and 8c05f781381d622ee7626c8978908d1b5fe03082

**Comment by ausLancaster** [comment]
> > Fixed bug where `<i>` tags were appearing with void voice (e.g. when wearing obsidian crown).
> 
> I believe this may have already been fixed
> 
> see #2517 and [8c05f78](https://github.com/goonstation/goonstation/commit/8c05f781381d622ee7626c8978908d1b5fe03082)

Ah you're right. I've removed my fix.

**Comment by ZeWaka** [comment]
Could you please label this PR?

File: code/mob/living.dm [hunk 1]
```diff
@@ -2013,3 +2014,16 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
...
 				src.was_harmed(thr.user, AM)
 	..()
+
+/mob/living/proc/check_singing_prefix(var/message)
+	if (isalive(src))
+		// check for "%"
+		if (dd_hasprefix(message, singing_prefix))
+			src.singing = NORMAL_SINGING
+			return copytext(message, 2)
+		// check for " %"
```
Comment (ZeWaka) [inline code/mob/living.dm:2024]: couldn't you just trim the message?

Comment (ausLancaster) [reply code/mob/living.dm:2024]: Could you give me an example of what you mean?

Comment (ausLancaster) [reply code/mob/living.dm:2024]: Do you mean using trim()? I think that would remove the "%" but I need to read that.

Comment (ZeWaka) [reply code/mob/living.dm:2024]: I think trim() just removes whitespace, unless I'm wrong...

Comment (ausLancaster) [reply code/mob/living.dm:2024]: Looks like it removes all characters with ascii code > 32
image

Comment (ausLancaster) [reply code/mob/living.dm:2024]: Oh I think you're right

Comment (ausLancaster) [reply code/mob/living.dm]: I've changed it so that say() is called before checks for '%'

**Comment by ausLancaster** [comment]
Is this good to merge?

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**